inference-engine/src/inference_engine/include/ie/ie_blob.h

```diff
@@ -242,6 +242,18 @@ class INFERENCE_ENGINE_API_CLASS(Blob) {
      */
     virtual Blob::Ptr createROI(const ROI& roi) const;
 
+    /**
+     * @brief Creates a blob describing given ROI object based on the current blob with memory sharing.
+     *
+     * Note: default implementation may throws "not implemented" exception.
+     *
+     * @param begin A ROI start coordinate
+     * @param end A ROI end coordinate
+     *
+     * @return A shared pointer to the newly created ROI blob.
+     */
+    virtual Blob::Ptr createROI(const std::vector<std::size_t>& begin, const std::vector<std::size_t>& end) const;
```

Comment:
BTW, since we already have `TensorSlice` struct, does it make sense to have `createROI` which accepts `std::vector<TensorSlice>`?

Reply:
`TensorSlice` is interanl API